src/openforms/submissions/logic/actions.py

```diff
@@ -225,16 +226,37 @@ class ServiceFetchAction(ActionOperation):
     def from_action(cls, action: ActionDict) -> "ServiceFetchAction":
         return cls(variable=action["variable"], fetch_config=action["action"]["value"])
 
-    def eval(self, context: DataMapping) -> DataMapping:
+    def eval(
+        self,
+        context: DataMapping,
+        log: Callable[[JSONValue], None],
+    ) -> DataMapping:
         # XXX perform_service_fetch is expressed in terms of FormVariables
         dummy_var = FormVariable(
             name=self.variable,
             service_fetch_configuration=ServiceFetchConfiguration.objects.get(
                 pk=self.fetch_config
             ),
         )
-        with log_errors({}, self.rule):  # TODO proper logging/error handling
-            return {dummy_var.name: perform_service_fetch(dummy_var, context)}
+        with log_errors({}, self.rule):  # TODO proper error handling
```

Comment:
This context manager is a bit to concrete.
I'm not sure whether I should change it, or turn this instance into a try-except.

We might even do a `log()` call on exception and log the error to the history in the admin.

Reply:
the admin `log` call makes sense, but it should be generic, something along the lines of "Service fetch action crashed (contact technical support)."

It kind of depends if it's a crash caused by bad logic rules, or by bugs by developers. The logic log page is meant for form designers so they can test their forms and see that their logic rules are behaving as they expect.